2017/proposals/yashchuk-ivan.md

```diff
@@ -0,0 +1,120 @@
+
+## Technical Details
+
+Many constituent parts to assemble and solve on quad/hex meshes are already in FEniCS, but
+there are missing links to get the pieces working as a whole.
+
+A key technological innovation in FEniCS is the development of a domain specific language
+for specifying finite element variational forms (UFL) and a form compiler (FFC) that can 
+translates UFL into low-level C++ code that is used to generate cell tensors (local matrices) on
+every cell in the mesh.
+
+Currently this toolchain cannot produce the C++ code for quad/hex cell geometries. What is needed to do is 
+to interface FFC with an existing class in FInite element Automatic Tabulator (FIAT) to evaluate the tensor product finite 
+element basis functions on the quad/hex cell geometry. An appropriate features
+are needed to be added into FFC to compute geometric quantities on the quad/hex cell geometry, using the existing
+code for triangles and tetrahedrons as an example. Finally, linking everything to the DOLFIN problem solving 
+environment should be done to solve a complete problem.
+
+The main source of information for this project is [The FEniCS Book](https://link.springer.com/book/10.1007%2F978-3-642-23099-8).
+Any other book on FEM can also be useful (for example a [book by Claes Johnson](https://www.amazon.com/Numerical-Solution-Differential-Equations-Mathematics/dp/048646900X))
+
+## Schedule of Deliverables
+
+### May 4th - May 30th, **Community Bonding Period**
+
+* Familiarize myself completely with FEniCS functionality and architecture.
+* Be in constant touch with my mentor and the FEniCS community.
+* Thus with the help of my mentor I will become absolutely clear about my future goals, the final implementations that need to be done as well as the approach that I will follow.
+
+### May 30th - June 3rd
+
+* Start working on building an interface between FFC and FIAT for quad/hex mesh.
+
+### June 5th - June 9th
+
+* Testing and improving the interface.
+
+### June 12th - June 16th
+
+* Work on quadrature representation for quad/hex mesh in FFC.
+
+### June 19th - June 23th
+
+* Work on tensor representation for quad/hex mesh in FFC.
+* Skeleton code for FIAT and FFC should be ready by Phase 2.
+
+### June 26 - June 30th, **End of Phase 1**
+
+* Existing code evaluation and improvement.
+
+### July 3rd - July 7th, **Begin of Phase 2**
+
+* Testing and bug fixing of the existing changes to FFC, FIAT code.
+
+### July 10th - July 14th
+
+* Finalize changes for quad/hex mesh in FIAT and FFC.
+
+### July 17th - July 21th
+
+* Start linking quad/hex mesh functionality to DOLFIN environment.
+
+### July 24th - July 28th, **End of Phase 2**
+
+* Existing code evaluation and improvement.
+
+### July 31st - August 4th, **Begin of Phase 3**
+
+* Continue working with DOLFIN.
+
+### August 7th - August 11th
+
+* Finalize linking everything to DOLFIN.
```

Comment:
You have sketched a fairly linear workflow where you first implement the FIAT/FFC interface, then update other parts of code generation in FFC, then link it up with DOLFIN. For the bulk of the work this is the right order of things, _but_ i think involving the entire pipeline including DOLFIN as early as possible will be more productive.